testsuite/results/Issue_452_pointers.opt

```diff
@@ -18,6 +18,10 @@
 	pop	bc
 	push	hl
 	push	de
+	ld	hl,0	;const
```

Comment:
Looks like there's a later optimisation that is removed by the new rules you've added

Reply:
How do I make mine run after?

Reply:
The only guaranteed way is to put them in a different file.

The rules run in this order (according the -O level):

```
                switch (peepholeopt) {
                case 0:
                    break;
                case 1:
                    rules[num_rules++] = c_coptrules1;
                    break;
                case 2:
                    rules[num_rules++] = c_coptrules2;
                    rules[num_rules++] = c_coptrules1;
                    break;
                default:
                    rules[num_rules++] = c_coptrules2;
                    rules[num_rules++] = c_coptrules1;
                    rules[num_rules++] = c_coptrules3;
                    break;
                }

    { "COPTRULES1", 0, SetStringConfig, &c_coptrules1, NULL, "", "\"DESTDIR/lib/z80rules.1\"" },
    { "COPTRULES2", 0, SetStringConfig, &c_coptrules2, NULL, "", "\"DESTDIR/lib/z80rules.2\"" },
    { "COPTRULES3", 0, SetStringConfig, &c_coptrules3, NULL, "", "\"DESTDIR/lib/z80rules.0\"" },
```

So we'd add an extra one to run after z80rules.1 in all cases.

Before making changes to the zcc driver, pull them into a separate file and then compile with `-custom-copt-rules=filename` and make sure you get the result you expect.

Reply:
Thanks, will look into it.

Reply:
Also, look at the code section without optimisation to see precisely which rule is (not) being triggered.

Perhaps the rule can be tweaked or reordered slightly relative to the new rules, which would be a light touch option to try?